explore/MultiTaskClassification.py

```diff
@@ -0,0 +1,308 @@
+genes = TT.set_index('Entrez_gene_id')['Gene_symbol'].to_dict()
+
+for k, v in genes.items():
+    print (k, ':', v)
+
+
+# ### Definition of functions and classifiers (mono and multitask)
+
+# In[7]:
+
+#functions for data preprocessing and figure labels
+
+def fs_mad(x, y):
+    """
+    Get the median absolute deviation (MAD) for each column of x
+    """
+    scores = mad(x) 
+    return scores, np.array([np.NaN]*len(scores))
+# select the top features with the highest MAD
+feature_select = SelectKBest(fs_mad, k=2000)
+
+pipeline = make_pipeline(
+    feature_select,  # Feature selection
+    StandardScaler(),  # Feature scaling
+    )
+
+def autolabel(rects,ay):
+    # attach some text labels
+    for rect in rects:
+        width = rect.get_width()
+        ay.text(1.05*width, rect.get_y() + rect.get_height()/2., 
+                '%.2f' % width,
+                ha='center', va='bottom')
+        
+
+
+# In[27]:
+
+# Mono-task classifiers, random forests and Lasso. Lasso results will be binarized. 
+
+Classifiers_mono = {
+    "Random forest": RandomForestClassifier(max_depth=100, min_samples_split=2, n_estimators=20, random_state=0),
+    "LassoCV": LassoCV(alphas=[.01, .1, 1, 10],tol=0.01),
+}
+
+
+# In[28]:
+
+# Following methods allow multi-task implementation: Extra trees, Knn, Random forests and MultiTask Lasso. 
+# Extra Tres and KNN will not be used in this notebook.
+
+Classifiers_multi = {
+    "Random forest": RandomForestClassifier(max_depth=100, min_samples_split=2, n_estimators=20, random_state=0),
+    "MultiTaskLassoCV": MultiTaskLassoCV(alphas=[.01, .1, 1, 10],tol=0.01),
+}
+
+
+# ### Training and testing dataset
+
+# In[29]:
+
+# all targeted genes, needed for comparison, store them in y
+
+y=Y[list(genes.keys())] 
+
+# Using same training set for all targets
+# Set aside 10% of the data for testing, this can only be done where the number of mutations is large enough
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
+'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))
+
+features_train=pipeline.fit_transform(X=X_train, y=y_train) 
+features_test=pipeline.transform(X=X_test)
+
+
+# ### Results for all genes
+
+# In[30]:
+
+# Results for all genes when observed separately, accuracy score and Auroc-s are used for evaluation
+
+as_RF =[]
+as_L = []
+
+for GENE in list(genes.keys()):
+    
+    y_train_each = y_train[GENE]
+    true_label = list(y_test[GENE])
+    
+    y_test_predict = dict()
+    for name, classifier in Classifiers_mono.items():
+        classifier.fit(features_train, y_train_each)
+        y_test_predict[name] = classifier.predict(features_test)
+        
+    
+    result_RF = y_test_predict['Random forest']
+    result_L = binarize(y_test_predict['LassoCV'].reshape(-1,1), threshold=0.5)     
+    as_RF.append(accuracy_score(true_label,result_RF))
+    as_L.append(accuracy_score(true_label,result_L))

```

Comment:
Accuracy score is not a good performance metric here, since our data is not balanced (we have more negatives than positives). For example, you can predict every sample as negative and get near 100% accuracy. You can use balanced accuracy instead. Or you can use AUROC, which I recommend since it accounts for multiple classification thresholds not just a single threshold.